Unable to compile gocsi

# Summary
Make fails with error:
service/service.go:26:9: cannot use &service literal (type *service) as type Service in return argument:
        *service does not implement Service (missing ControllerExpandVolume method)


# New Feature
No

# Bug Reports
This section is for issues that relate to discovered problems or bugs.

## Version
Please include the version of GoCSI and the CSI API model.
Latest: 0.5.0?
## Expected Behavior
Please describe in detail the expected behavior.
Make should succeed without any error

## Actual Behavior
Please describe in detail the actual behavior.

## Steps To Reproduce
Please list the steps to reproduce the issue in this section.

test@dsib1241:~/go/src$ docker run -it golang:latest sh -c \
  "go get github.com/rexray/gocsi && \
  make -C src/github.com/rexray/gocsi csi-sp"


* Please list each step with as much detail as possible.
* The more information gathered up front, the easier it is to solve
the problem.
* Thank you!
OS: Ubuntu 16.04

## Logs
test@dsib1241:~/go/src$ docker run -it golang:latest sh -c \
>   "go get github.com/rexray/gocsi && \
>   make -C src/github.com/rexray/gocsi csi-sp"
Unable to find image 'golang:latest' locally
latest: Pulling from library/golang
22dbe790f715: Pull complete
0250231711a0: Pull complete
6fba9447437b: Pull complete
c2b4d327b352: Pull complete
619f4932b7ea: Pull complete
592933ec07f7: Pull complete
19003e5a1e3a: Pull complete
Digest: sha256:709b2f45821faee19a90494838706d90bbc5a50e9fb0501c1a43a006e9654a80
Status: Downloaded newer image for golang:latest
make: Entering directory '/go/src/github.com/rexray/gocsi'
git clone https://github.com/akutz/dep ./vendor/github.com/golang/dep
Cloning into './vendor/github.com/golang/dep'...
remote: Enumerating objects: 19210, done.
remote: Total 19210 (delta 0), reused 0 (delta 0), pack-reused 19210
Receiving objects: 100% (19210/19210), 11.70 MiB | 8.05 MiB/s, done.
Resolving deltas: 100% (11398/11398), done.
git -C ./vendor/github.com/golang/dep checkout feature/fetch-gh-pulls
Branch feature/fetch-gh-pulls set up to track remote branch feature/fetch-gh-pulls from origin.
Switched to a new branch 'feature/fetch-gh-pulls'
go build -o dep ./vendor/github.com/golang/dep/cmd/dep
DEP=/go/src/github.com/rexray/gocsi/dep USE_DEP=true DEP_GIT_REFSPECS=+refs/pull/*:refs/pull/origin/* ./gocsi.sh csi-sp
creating project directories:
  /go/src/csi-sp
  /go/src/csi-sp/provider
  /go/src/csi-sp/service
creating project files:
  /go/src/csi-sp/main.go
  /go/src/csi-sp/provider/provider.go
  /go/src/csi-sp/service/service.go
  /go/src/csi-sp/service/controller.go
  /go/src/csi-sp/service/identity.go
  /go/src/csi-sp/service/node.go
using golang/dep:
  executing dep ensure
building csi-sp:
# csi-sp/service
service/service.go:26:9: cannot use &service literal (type *service) as type Service in return argument:
        *service does not implement Service (missing ControllerExpandVolume method)
Makefile:89: recipe for target '/go/src/csi-sp/csi-sp' failed
make: *** [/go/src/csi-sp/csi-sp] Error 1
make: Leaving directory '/go/src/github.com/rexray/gocsi'

--- a/.gitignore
+++ b/.gitignore
@@ -12,6 +12,9 @@
 /etcd.log
 default.etcd
 
+# The csi-sp test log file
+csi-sp.log
+
 # The archives built from this package.
 /*.a
 /middleware/*.a

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,7 +9,7 @@ dist: trusty
 
 # Select Go as the language used to run the buid.
 language: go
-go: 1.9.4
+go: 1.12
 go_import_path: github.com/rexray/gocsi
 
 jobs:

--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -14,7 +14,7 @@
 
 [[constraint]]
   name = "github.com/container-storage-interface/spec"
-  version = "1.0.0"
+  version = "=1.0.0"
 
 [[constraint]]
   name = "github.com/golang/protobuf"

--- a/Makefile
+++ b/Makefile
@@ -86,6 +86,8 @@ GOCSI_SH_ENV += GOCSI_DEP_SOURCE=https://github.com/$(TRAVIS_REPO_SLUG)
 GOCSI_SH_ENV += GOCSI_DEP_REVISION=$(TRAVIS_COMMIT)
 endif
 $(CSI_SP): | $(DEP)
+	mkdir -p $(CSI_SP_DIR)
+	cp Gopkg.toml $(CSI_SP_DIR)
 	DEP=$(abspath $(DEP)) $(GOCSI_SH_ENV) ./gocsi.sh $(CSI_SP_IMPORT)
 
 csi-sp: $(CSI_SP_LOG)

